cabal-install/src/Distribution/Client/CmdInstall.hs

```diff
@@ -440,11 +440,12 @@ installAction flags@NixStyleFlags{extraFlags = clientInstallFlags', ..} targetSt
 
         ProjectConfigShared
           { projectConfigStoreDir
+          , projectConfigPackageDBs
           } = projectConfigShared projectConfig
 
         mlogsDir = flagToMaybe projectConfigLogsDir
         mstoreDir = flagToMaybe projectConfigStoreDir
-      cabalDirLayout <- mkCabalDirLayout mstoreDir mlogsDir
+      cabalDirLayout <- mkCabalDirLayout mstoreDir mlogsDir projectConfigPackageDBs
```

Comment:
This should raise eyebrows. How come the CABAL_DIR layout now depends on project level configuration?

Reply:
Because the initial package database stack depends on the `package-db` option, which is specified at a project level.

This seems completely natural to me, and changing the implementation like this caught the fact that `--package-db` flag was not honoured by the `install` command. What is your proposed alternative?

Reply:
The layout of CABAL_DIR does not depend on any project configuration. What happens here is that `CabalDirLayout` contains (through `StoreDirLayout`)
```
storePackageDBStack :: Compiler -> PackageDBStack
storePackageDBStack compiler = [GlobalPackageDB, storePackageDB compiler]
```
You want to insert project packagedbs inside that stack and you cannot do it. IMHO the right solution is either

1) Change that function to take a list of packagedb (just like it takes a compiler):
```
storePackageDBStack :: Compiler -> [PackageDB] -> PackageDBStack
storePackageDBStack compiler packagedbs = GlobalPackageDB : packagedbs ++ [storePackageDB compiler]
```
(modulo a `Just`, I think we should only allow "real" packagedbs rather than "maybe encoded packagedb commands")

or

2) Remove that function entirely and rely on `storePackageDB :: Compiler -> PackageDB` instead. The accessor `storePackageDBStack` does not really save any code and you just have proven that it was hiding an important decision (the order).

Reply:
That is a good idea to change the function to take an extra argument. I have done that.